database/factories/CommentFactory.php

```diff
@@ -3,25 +3,35 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
 // See the LICENCE file in the repository root for full licence text.
 
-$factory->define(App\Models\Comment::class, function (Faker\Generator $faker) {
-    return [
-        'user_id' => function () {
-            return App\Models\User::factory()->create()->user_id;
-        },
-        'message' => function () use ($faker) {
-            return $faker->paragraph();
-        },
-        'commentable_type' => function () {
-            return 'build'; // TODO: add support for more types
-        },
-        'commentable_id' => fn () => App\Models\Build::factory()->create()->getKey(),
-        'created_at' => now(),
-        'updated_at' => now(),
-    ];
-});
-
-$factory->state(App\Models\Comment::class, 'deleted', function () {
-    return [
-        'deleted_at' => now(),
-    ];
-});
+namespace Database\Factories;
+
+use App\Libraries\MorphMap;
+use App\Models\Comment;
+use App\Models\User;
+use Illuminate\Database\Eloquent\Factories\Factory;
+
+class CommentFactory extends Factory
+{
+    protected $model = Comment::class;
+
+    public function definition(): array
+    {
+        return [
+            'commentable_type' => fn () => $this->faker->randomElement(Comment::COMMENTABLES),
+            'created_at' => fn () => $this->faker->dateTime(),
```

Comment:
nothing else use random date for `created_at` and `updated_at` and I don't think it's needed either (also for NewsPost)